### PR TITLE
PPC: fixed compilation warnings

### PR DESCRIPTION
**Pull request by hoopoepg**

### CODE REVIEW

File: src/ucs/sys/sock.c [hunk 1]
```diff
@@ -334,11 +335,18 @@ ucs_status_t ucs_socket_send(int fd, const void *data, size_t length,
                               "send", err_cb, err_cb_arg);
 }
 
+/* recv is declared as 'always_inline' on some platforms, it leads to
+ * compilation warning. wrap it into static function */
+static ssize_t ucs_socket_recv_io(int fd, void *data, size_t size, int flags)
```
Comment (brminich) [inline src/ucs/sys/sock.c]: is it possible to suppress this warning somehow instead?

Comment (hoopoepg) [reply src/ucs/sys/sock.c]: specific command line option to suppress warning... but this is not good idea

Comment (brminich) [reply src/ucs/sys/sock.c]: it is used in `uct_tcp_ep_recv` which is invoked by TCP progress
should not affect perf if `recv` is declared inline though

**Comment by brminich** [comment]
btw, why did u update just `ucs_socket_recv`?
what about `ucs_socket_recv_nb` and others?

**Comment by hoopoepg** [comment]
because uct_tcp_ep_recv is reported in error

**Comment by brminich** [comment]
> because uct_tcp_ep_recv is reported in error

Can you please check other places and fix them accordingly?
there are similar issues, like described in #4592

**Comment by hoopoepg** [comment]
any ideas how to reproduce? :)

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 25-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/0e7d86c7356de9a41494d35e741d04b1d9dd9b3b/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/88957cb38a83c4b60bff463c966ec4c0cbc44995/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/79d6db3486099e2dbf24634a5065ab6e1cfb8ba1/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/d82b940c5e78eeabba6f42b4d27392d48af4972a/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/e38fdee3078cec94cbebd7668ead3cb4c63f2f6e/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/35b031b3d6a5177c0285294ec95b7d3a5e183819/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/c1d0ea2ce9093501838c35ebf0250ed7a239cc01/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/f69c8d7aee18f677427def631cd9a151d0ce8d64/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/cc4c5ebe896eaafe239b85a76bf042578715e4eb/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/2f8f4fd2b13c9f91755a8ae8d40409c26f5ab968/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/4a38dd92cae19531d671568dd050ad8925ccb59e/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/7f20695ef5e0331df54d3a0f5924b099fdda3bd8/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/276f218e9c3de93881be90f91e9b95e0751c0468/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/7f3254a9dc9a30ab82f4af0cb2de24b7c1313085/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/b394bf886de95e9836487f54407a9deeb7e5db41/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/975d30b659451a5091b990145203e2b5dc59b142/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/13e29b085f24cdeee214aa4e96cf62351f802a54/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/d19730b9b9977173abef7ab7f6c6e447b4787dae/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/cd0caba2e1df3a7d81f087f048e92ea162ba5fc4/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/87d77a254f485c65e52b27beda8cde2c17bd4d6e/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/9e569586d615999a6ec83dbc5aa6f35bae3beda6/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/556a4b04dc75a8feb876a4eb9d4f5951c0a9fa8b/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/4e5c2e004f391dcaf3e3107184dca7cc725e4e70/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/ae4326b96e391479d31d51cf9a6db03706095473/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/c3f3d41a91d7a303be7c7938d71cced1/raw/0a3ce6847f8bc07c03c38f9c5aa2216bcd8ea4e8/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**